### PR TITLE
Microsoft.IoT.Lightning.Providers initial commit

### PR DESCRIPTION
**Pull request by MahmoudGSaleh**

### CODE REVIEW

File: Microsoft.IoT.Lightning.Providers/Blinky/Blinky.csproj [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    <Platform Condition=" '$(Platform)' == '' ">x86</Platform>
+    <ProjectGuid>{F89AF804-259E-4AEB-9F0D-C6D7ACEA7529}</ProjectGuid>
+    <OutputType>AppContainerExe</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>Blinky</RootNamespace>
+    <AssemblyName>Blinky</AssemblyName>
+    <DefaultLanguage>en-US</DefaultLanguage>
+    <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
+    <TargetPlatformVersion>10.0.10556.0</TargetPlatformVersion>
+    <TargetPlatformMinVersion>10.0.10240.0</TargetPlatformMinVersion>
```
Comment (bfjelds) [inline Microsoft.IoT.Lightning.Providers/Blinky/Blinky.csproj]: 10240 is RTM, can the dmap providers be used with an RTM SDK???

File: Microsoft.IoT.Lightning.Providers/Blinky/Package.appxmanifest [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+<Package
+  xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
+  xmlns:mp="http://schemas.microsoft.com/appx/2014/phone/manifest"
+  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
+  xmlns:iot="http://schemas.microsoft.com/appx/manifest/iot/windows10"
+  IgnorableNamespaces="uap mp iot">
+
+  <Identity
+    Name="ce88ee3f-c2d3-4148-9fd4-73d6a075b6b5"
+    Publisher="CN=msaleh"
```
Comment (bfjelds) [inline Microsoft.IoT.Lightning.Providers/Blinky/Package.appxmanifest]: remove msaleh references

**Comment by bfjelds** [comment]
remove all project.lock.json files

**Comment by bfjelds** [comment]
be sure to include all sln files (i didn't see one for any blinky\* projects)

**Comment by bfjelds** [comment]
be sure you have LICENSE.txt in all appropriate places

File: Microsoft.IoT.Lightning.Providers/Blinky/MainPage.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
+﻿using System;
```
Comment (bfjelds) [inline Microsoft.IoT.Lightning.Providers/Blinky/MainPage.xaml.cs]: add copyright comment to all source code

Comment (ooeygui) [inline Microsoft.IoT.Lightning.Providers/Blinky/MainPage.xaml.cs]: New guidance from LCA is that top level license.txt is all that is required - each file need not have a copyright and licence header.

**Comment by MahmoudGSaleh** [comment]
Thanks! The PR now has the following changes:
1. Removed references to SDK 10240, and added 10556, min needed for deployment.
2. Replaced all user references with Microsoft
3. project.lock.json files removed
4. Removed top level .sln file
5. Added new .sln files, one for the provider library and for each of the samples
6. LICENSE.txt added everywhere, top level and in each of the solutions

File: Microsoft.IoT.Lightning.Providers/Providers/Microsoft.Iot.Lightning.Providers.vcxproj [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+    <WholeProgramOptimization>true</WholeProgramOptimization>
+    <PlatformToolset>v140</PlatformToolset>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
+  <ImportGroup Label="ExtensionSettings">
+  </ImportGroup>
+  <ImportGroup Label="Shared">
+  </ImportGroup>
+  <ImportGroup Label="PropertySheets" Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
```
Comment (bfjelds) [inline Microsoft.IoT.Lightning.Providers/Providers/Microsoft.Iot.Lightning.Providers.vcxproj:82]: what are all of these imports?

File: Microsoft.IoT.Lightning.Providers/Providers/Microsoft.Iot.Lightning.Providers.vcxproj.filters [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    </Filter>
+  </ItemGroup>
+  <ItemGroup>
+    <ClCompile Include="pch.cpp" />
+    <ClCompile Include="I2cDeviceProvider.cpp" />
+    <ClCompile Include="GpioDeviceProvider.cpp" />
+    <ClCompile Include="Provider.cpp" />
+    <ClCompile Include="SpiDeviceProvider.cpp" />
+    <ClCompile Include="Dummy.cpp" />
+    <ClCompile Include="@(ArduinoSDKSources)" />
```
Comment (bfjelds) [inline Microsoft.IoT.Lightning.Providers/Providers/Microsoft.Iot.Lightning.Providers.vcxproj.filters:16]: why are there 6 clcompile items for arduinosdksrouces?

**Comment by bfjelds** [comment]
remove all project.lock.json files

**Comment by bfjelds** [comment]
check with jesse, but for the sample aggregate provider i'm working on, jesse suggested that the only public constructor be the aggregate constructor.

File: Microsoft.IoT.Lightning.Providers/Providers/Microsoft.Iot.Lightning.Providers.vcxproj.filters (hunk 5 above)
Comment (MahmoudGSaleh) [inline Microsoft.IoT.Lightning.Providers/Providers/Microsoft.Iot.Lightning.Providers.vcxproj.filters:16]: Looks like they get added by VC++ when referencing these sources from the targets of the nuget packages.

**Comment by MahmoudGSaleh** [comment]
Still waiting on the final SDK version to include.

**Pull request closed without merging**